src/main/java/org/junitpioneer/jupiter/DefaultLocaleExtension.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.util.Locale;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+class DefaultLocaleExtension implements BeforeAllCallback, BeforeEachCallback, AfterAllCallback, AfterEachCallback {
+
+	private static final Namespace NAMESPACE = Namespace.create(DefaultLocaleExtension.class);
+
+	private static final String KEY = "DefaultLocale";
+
+	@Override
+	public void beforeAll(ExtensionContext context) {
+		setDefaultLocale(context);
+	}
+
+	@Override
+	public void beforeEach(ExtensionContext context) {
+		if (annotationPresentOnTestMethod(context)) {
+			setDefaultLocale(context);
```

Comment:
Note that the invocation of both of these methods actually results in the annotation being looked up twice, which seems a bit odd to me. Though I suppose it's not much of a performance hit. 😇

Reply:
We can have `annotationPresentOnTestMethod` return an `Optional<$whatever>` and then act on that. :smiley:

Reply:
Created #115 to implement this optimization